### PR TITLE
issue46 - Support for AndroidTV. extend ActionBarActivity for compatibility with Android TV.

### PR DESCRIPTION
**Pull request by cannino**
I followed the [Getting Started with TV Apps](https://developer.android.com/training/tv/start/start.html) page and used Android Studio v3.0.1.  Some of these changes are a result of using a newer environment and some are recommendations based on the Getting Started with TV Apps link. I'll try and highlight each of those in the changes below.

### CODE REVIEW

File: AndroidManifest.xml [hunk 1]
```diff
@@ -16,18 +16,19 @@
 -->
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.oakesville.mythling"
-    android:versionCode="48"
-    android:versionName="2.1.13" >
+    android:versionCode="49"
```
Comment (cannino) [inline AndroidManifest.xml]: should I update the version and build or leave that to maintainers?

File: AndroidManifest.xml [hunk 2]
```diff
@@ -16,18 +16,19 @@
...
     package="com.oakesville.mythling"
-    android:versionCode="48"
-    android:versionName="2.1.13" >
+    android:versionCode="49"
+    android:versionName="2.1.14" >
 
     <uses-sdk
-        android:minSdkVersion="14"
-        android:targetSdkVersion="15" />
+        android:minSdkVersion="21"
```
Comment (cannino) [inline AndroidManifest.xml]: min recommendation from "getting started with tv apps" link

File: AndroidManifest.xml [hunk 3]
```diff
@@ -41,6 +42,7 @@
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
                 <category android:name="android.intent.category.LAUNCHER" />
+                <category android:name="android.intent.category.LEANBACK_LAUNCHER" />
```
Comment (cannino) [inline AndroidManifest.xml]: also recommendation from "getting started with tv apps" link. I didn't see any reason to create a new intent-filter that would call the same action, so I added a new category here. Schema seems to support multiple category elements.

File: build.gradle [hunk 4]
```diff
@@ -43,7 +46,7 @@ android {
     }
 
     dexOptions {
-        preDexLibraries = false
+        preDexLibraries false
```
Comment (cannino) [inline build.gradle]: if i remember right when upgrading build tools i got an error here when using equals.

File: build.gradle [hunk 5]
```diff
@@ -6,19 +6,22 @@ buildscript {
...
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.1.3'
+        classpath 'com.android.tools.build:gradle:3.0.1'
     }
 }
 
 dependencies {
-    compile fileTree(dir: 'libs', include: '*.jar')
+    implementation fileTree(include: '*.jar', dir: 'libs')
```
Comment (cannino) [inline build.gradle]: gradle upgrade uses 'implementation' now instead of compile

File: build.gradle [hunk 6]
```diff
@@ -6,19 +6,22 @@ buildscript {
     repositories {
         mavenCentral()
 		jcenter()
+        maven {
```
Comment (cannino) [inline build.gradle]: hmmm, may not need this....

File: gradle/wrapper/gradle-wrapper.properties [hunk 7]
```diff
@@ -1,6 +1,6 @@
...
-distributionPath=wrapper/dists
-zipStoreBase=GRADLE_USER_HOME
-zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
+#Sun Dec 10 01:57:57 MST 2017
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
```
Comment (cannino) [inline gradle/wrapper/gradle-wrapper.properties]: gradle upgrade, rest of diff is whitespace changes as I am working on a Mac

File: res/values/styles.xml [hunk 8]
```diff
@@ -16,7 +16,7 @@
 -->
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
 
-    <style name="MythlingTheme" parent="android:Theme.Holo.Light">
+    <style name="MythlingTheme" parent="@style/Theme.AppCompat.Light">
```
Comment (cannino) [inline res/values/styles.xml]: see https://github.com/oakesville/mythling/issues/46#issuecomment-352215719

File: src/com/oakesville/mythling/MainActivity.java [hunk 9]
```diff
@@ -64,7 +64,7 @@ public void onClick(DialogInterface dialog, int which) {
             }).show();
         }
 
-        setContentView(getAppSettings().isTv() ? R.layout.firetv_split : R.layout.split);
+        setContentView(getAppSettings().isFireTv() ? R.layout.firetv_split : R.layout.split);
```
Comment (cannino) [inline src/com/oakesville/mythling/MainActivity.java]: not sure I need this, but found it interesting isTv was being checked but a instance of firetv_split was used in the tertiary expression.

File: src/com/oakesville/mythling/prefs/PrefsActivity.java [hunk 10]
```diff
@@ -80,6 +80,11 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return super.onOptionsItemSelected(item);
     }
 
+    @Override
+    protected boolean isValidFragment(String fragmentName) {
```
Comment (cannino) [inline src/com/oakesville/mythling/prefs/PrefsActivity.java]: got a runtime exception after upgrading to sdk 21. I am just returning true here to satisfy an 'unimplemented method' error. When debugging the fragmentName parameter, the value was "com.oakesville.mythling.prefs.NetworkPrefs".

I don't have anything valuable to 'validate' here basically so just return 'true'

**Comment by oakesville** [comment]
Thanks for the PR.  I'll try and merge this weekend.  I want to give some thought to minSdkVersion.  The Google Play dev console tells me that somewhere around 14% of active mythling users are on a version older than Android 5.0.  Does it run okay for you with minSdkVersion="19"?

**Comment by cannino** [comment]
Sure Don, I'll try that out today and let you know the results.

**Comment by cannino** [comment]
No problems with minSDK 19 on my TV. I fixed some other places we are not calling getSupportActionBar which caused an NPE when loading the help menu.

**I should have noted  before that the action bar menu seems to change the layout. Instead of getting a 'Recordings',  'Sort Type', '...' all you get now is a '...' which you can click and then have access to the full menu. It does involve an extra click which I am not sure will be acceptable, as I sort like the old menu better. This also affects when running on a phone.**

**Pull request merged**